### PR TITLE
Bug: Fix ability to load if automatically loaded page doesn't fill entire screen

### PR DESCRIPTION
**Pull request by mikedg1**
Description:
If a page of posts doesn't have enough non-hidden posts to fill the screen, posts will not be loaded when you scroll since the onAppear for that view won't happen again. This can happen if you have "hide read posts" enabled.

Changes Made:
Added UUID to the view that triggers additional loading. This seems to fix the issue. A lot of changes were made to arrive at the simple approach, so definitely squash merge this.

Screenshots:
Code used to force the issue.
image

Shows how it works with fix:
https://github.com/user-attachments/assets/ad3df228-8a3e-4cf9-b3d7-c1d9ae327fb1

Shows how it works prior to fix:

https://github.com/user-attachments/assets/700b8dae-0321-4757-9bbe-b1ed7db59b75

### CODE REVIEW

File: OpenArtemis.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1009,7 +1009,7 @@
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 1;
 				DEVELOPMENT_ASSET_PATHS = "\"OpenArtemis/Preview Content\"";
-				DEVELOPMENT_TEAM = 2Q775S63Q3;
```
Comment (mikedg1) [inline OpenArtemis.xcodeproj/project.pbxproj:1012]: TODO @mikedg1 remove

File: OpenArtemis/Scrape/ScrapeSubreddit.swift [hunk 2]
```diff
@@ -43,7 +43,7 @@ class RedditScraper {
             completion(.failure(NSError(domain: "Invalid URL", code: 0, userInfo: nil)))
             return
         }
-
```
Comment (mikedg1) [inline OpenArtemis/Scrape/ScrapeSubreddit.swift:46]: TODO: @mikedg1 remove

File: OpenArtemis/Views/Subreddits/SubredditFeedView.swift [hunk 3]
```diff
@@ -177,14 +179,14 @@ struct SubredditFeedView: View {
         if searchTerm.isEmpty {
             RedditScraper.scrapeSubreddit(subreddit: subredditName, lastPostAfter: lastPostAfter, sort: sort,
                                           trackingParamRemover: trackingParamRemover, over18: over18) { result in
-                defer {
+                defer { // what is this
                     isLoading = false
                 }
```
Comment (mikedg1) [inline OpenArtemis/Views/Subreddits/SubredditFeedView.swift:184]: TODO @mikedg1 remove all

File: OpenArtemis/Views/Utility/ScrollLoaderView.swift [hunk 4]
```diff
@@ -0,0 +1,41 @@
+////
+////  ScrollLoaderView.swift
```
Comment (mikedg1) [inline OpenArtemis/Views/Utility/ScrollLoaderView.swift:2]: @mikedg1 delete

File: OpenArtemis/Views/Subreddits/SubredditFeedView.swift [hunk 5]
```diff
@@ -74,10 +75,11 @@ struct SubredditFeedView: View {
                             }
                         }
                     }
-                    
+
                     Rectangle()
                         .fill(Color.clear)
                         .frame(height: 1)
+                        .id(UUID()) // adding this causes onAppear to be called multiple times even if the view didn't leave the screen
```
Comment (mikedg1) [inline OpenArtemis/Views/Subreddits/SubredditFeedView.swift:82]: Keep this

**Comment by mikedg1** [comment]
Work in progress, will be cleaned up tonight

**Comment by mikedg1** [comment]
Finished clean up

**Comment by ejbills** [comment]
Is this ready to review? :)

**Comment by mikedg1** [comment]
Yup, ready to review. 

Basically, we wanted onAppear to be called if the loading view moves downward but isn't pushed off the screen. I'm not experienced enough with SwiftUI to say why this works. I had originally come up with a solution that used an ID that just counted upward, then while working on another PR noticed you had some similar behavior elsewhere and tried just using UUID and it worked.

File: OpenArtemis/Views/Subreddits/SubredditFeedView.swift (hunk 5 above)
Comment (ejbills) [reply OpenArtemis/Views/Subreddits/SubredditFeedView.swift:81]: Is the comment here true, that it is triggering the on appear multiple times?

**Comment by ejbills** [comment]
I see, so once a page with read posts loaded, the loading trigger wouldn't execute the onAppear. Looks like adding the UUID spawns a new instance of the rectangle, thus allowing it to trigger the on appear again. (at least that's my guess)

**Pull request merged**